### ISSUE TITLE
Custom Buttons and Labels for QuickPicks

### ISSUE THREAD

**Issue by nturinski** [microsoft/vscode#139249]
It would be great to be able to add some custom buttons to our QuickPicks, especially for our Azure Extension wizards.  Specifically, being able to have an "OK" button would be useful so that the user can confirm without clicking on a quick pick item.

Here is an example of what I am referring to:
![image](https://user-images.githubusercontent.com/5290572/146286726-85f15441-d324-45ba-ac8a-95f5b7973bfa.png)

I think that the simpleFileDialog already leverages some unexposed properties to do this already.
https://github.com/microsoft/vscode/blob/91b97cec911a954bcd83d2613bb6d7b6bd4023ad/src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts#L280-L281

@isidorn

**Comment by TylerLeonhardt** [comment]
We have buttons in the title bar and in each individual item:
![image](https://user-images.githubusercontent.com/2644648/146445996-40d89a9b-d5a5-46b6-a2de-89de56cafaa5.png)

I would love to empower extension authors to do whatever they want to the quick pick but I need to know more reasoning for this beyond an "OK" button.

> so that the user can confirm without clicking on a quick pick item

The user could also use their keyboard, of course. Which would be an alternative to clicking on a specific item.

**Comment by nturinski** [comment]
Our use case is very similar to the `SimpleFileDialog`. We want users to be able to click the folder quickpick to drill deeper into the directory, but the OK button would be how they make their final confirmation.

**Comment by TylerLeonhardt** [comment]
Fair enough. I guess in the meantime you could use a ✅ button until an API for this is available.
